src/App.tsx

```diff
@@ -1,63 +1,97 @@
-import React, { useEffect, useState } from 'react';
+import { useEffect, useState } from 'react';
+import { BallTriangle } from 'react-loader-spinner';
 import { Route, Routes } from 'react-router-dom';
 
+import { clsx } from 'clsx';
+
 import About from '@/components/about/About.component.tsx';
+import Error from '@/components/Error/Error.components.tsx';
 import LayoutComponent from '@/components/LayoutComponent/LayoutComponent.tsx';
 import { useResize } from '@/components/myHooks/use-resize';
 import PageNotFound from '@/components/pageNotFound/PageNotFound.components.tsx';
 import ProductList from '@/components/productList/ProductList.component.tsx';
 import ProductPageComponent from '@/components/ProductPage/ProductPage.component.tsx';
 import { PagesContext } from '@/context/PagesContext.tsx';
+import type { CategoryName } from '@/interface';
 import type Product from '@/interface/product.ts';
 import getData from '@/utils/getData.ts';
 
 import './App.css';
 
+interface Category {
+    electronics: number;
+    shoes: number;
+    clothes: number;
+}
+
 const App = () => {
     let defaultTheme: string = window.matchMedia('(prefers-color-scheme: dark)').matches ? 'dark' : 'light';
     if (localStorage.getItem('theme')) {
         defaultTheme = localStorage.getItem('theme') === 'light' ? 'light' : 'dark';
     }
 
+    const categoryId: Category = {
+        electronics: 2,
+        shoes: 4,
+        clothes: 1,
+    };
+
     const [amountCart, setAmountCart] = useState<number>(0);
     const [allProducts, setAllProducts] = useState<Product[]>([]);
-    const [filterProducts, setFilterProducts] = useState<Product[]>([]);
-    const [products, setProducts] = useState<Product[]>([]);
     const [theme, setTheme] = useState<string>(defaultTheme);
-    const [step, setStep] = useState<number>(10);
-
+    const [step, setStep] = useState<number>(0);
     const [pagination, setPagination] = useState<{ start: number; end: number }>({ start: 0, end: step });
     const [totalPages, setTotalPages] = useState<number>(0);
     const [numberPage, setNumberPage] = useState<number>(1);
+    const [totalProducts, setTotalProducts] = useState<number>(0);
+    const [idCategory, setIdCategory] = useState<number>(0);
+    const [nameCategory, setNameCategory] = useState<string>('');
+    const [searchValue, setSearchValue] = useState<string>('');
+    const [isLoader, setIsLoader] = useState<boolean>(false);
+    const [errorMessage, setErrorMessage] = useState<string>('');
 
     const changeTheme = (name: string) => {
         name === 'light' ? setTheme('light') : setTheme('dark');
         localStorage.setItem('theme', name === 'light' ? 'light' : 'dark');
     };
```

Comment:
maybe you can write more simple? it's hart to understand, better do refactoring for this.

Reply:
fixed